nomic/project.py

```diff
@@ -513,32 +513,31 @@ def vector_search(self, queries: np.array = None, ids: List[str] = None, k: int
             bytesio = io.BytesIO()
             np.save(bytesio, queries)
 
-        status = 0
-        retries = 5
-        while status != 200 and retries >= 0:
-            if queries is not None:
-                response = requests.post(
-                    self.project.atlas_api_path + "/v1/project/data/get/nearest_neighbors/by_embedding",
-                    headers=self.project.header,
-                    json={'atlas_index_id': self.atlas_index_id,
-                          'queries': base64.b64encode(bytesio.getvalue()).decode('utf-8'),
-                          'k': k},
-                )
-            else:
-                response = requests.post(
-                    self.project.atlas_api_path + "/v1/project/data/get/nearest_neighbors/by_id",
-                    headers=self.project.header,
-                    json={'atlas_index_id': self.atlas_index_id,
-                          'datum_ids': ids,
-                          'k': k},
-                )
+        if queries is not None:
+            response = requests.post(
+                self.project.atlas_api_path + "/v1/project/data/get/nearest_neighbors/by_embedding",
+                headers=self.project.header,
+                json={'atlas_index_id': self.atlas_index_id,
+                      'queries': base64.b64encode(bytesio.getvalue()).decode('utf-8'),
+                      'k': k},
+            )
+        else:
+            response = requests.post(
+                self.project.atlas_api_path + "/v1/project/data/get/nearest_neighbors/by_id",
+                headers=self.project.header,
+                json={'atlas_index_id': self.atlas_index_id,
+                      'datum_ids': ids,
+                      'k': k},
+            )
 
-            status = response.status_code
-            retries -= 1
+        status = response.status_code
 
-        if retries == 0:
+        if response.status_code == 500:
```

Comment:
would love some logging here as well

Reply:
It throws an exception, log message would be redundant.